### ISSUE TITLE
aws_db_parameter_group error when more than 20 parameters

### ISSUE THREAD

**Issue by bramswenson** [hashicorp/terraform#1710]
I have a terraform file with 3 database parameter groups, each with over 20 parameters. All 3 fail to create with the following errors:

```
* Error modifying DB Parameter Group: Cannot modify more than 20 parameters in a single request
aws_db_parameter_group.production-master: Error: 1 error(s) occurred:

* Error modifying DB Parameter Group: Cannot modify more than 20 parameters in a single request
aws_db_parameter_group.production-follower: Error: 1 error(s) occurred:

* Error modifying DB Parameter Group: Cannot modify more than 20 parameters in a single request
Error applying plan:

3 error(s) occurred:

* 1 error(s) occurred:

* 1 error(s) occurred:

* Error modifying DB Parameter Group: Cannot modify more than 20 parameters in a single request
* 1 error(s) occurred:

* 1 error(s) occurred:

* Error modifying DB Parameter Group: Cannot modify more than 20 parameters in a single request
* 1 error(s) occurred:

* 1 error(s) occurred:

* Error modifying DB Parameter Group: Cannot modify more than 20 parameters in a single request

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

**Comment by bramswenson** [comment]
I suspect the code needs to queue more than one call to the api when there are more than 20 parameters: 
http://docs.aws.amazon.com/cli/latest/reference/rds/modify-db-parameter-group.html

**Comment by bramswenson** [comment]
I was able to trim my parameter group to exactly 20 and it works perfectly. I believe this verifies the issue.

**Comment by jonhatalla** [comment]
encountering this as well - I have 24 parameters to pass.  If i temporarily remove 4 of them, run plan and apply, then add the 4 parameters back.  And rerun the command, it will work.

Agreed with @bramswenson - a queuing system where it sends the parameters in multiple chunks (at most 20 parameters at a time) instead of all at once - would be a simple fix.  If i get time later this week, i'll try and create a PR.

**Comment by bramswenson** [comment]
Thanks @jonhatalla! I imagine such a pattern has already been needed in terraform somewhere. This isn't an uncommon type of restriction on aws apis.

**Comment by mikeyhill** [comment]
+1 - just ran into this issue as well, we have 30~ parameters on one of our RDS instances.

**Comment by Hasgaroth** [comment]
Me too, I have over 20 parameters and get the same error
It would be good if terraform would split the update into multiple groups of 20 for uat :)

**Comment by jlspies1** [comment]
We are also having this issue, would be great to get a plan in place to resolve this.

**Comment by whiteadam** [comment]
+1

**Comment by rbdixon** [comment]
+1

**Comment by phundisk** [comment]
+1

**Comment by bigkraig** [comment]
+1

**Comment by hSATAC** [comment]
+1

**Comment by liangyungong** [comment]
+1

**Comment by cazorla19** [comment]
+1

**Comment by stack72** [comment]
Closed via #7364

**Comment by dgarstang** [comment]
I don't believe this is closed.

```
resource "aws_elasticache_parameter_group" "cache" {
  name   = "${var.service_name}-${var.stack_id}"
  family = "redis3.2"
  /* These are verbatim of the default values. */
  parameter { name  = "activerehashing"                                value = "yes" }
  parameter { name  = "appendfsync"                                    value = "everysec" }
  parameter { name  = "appendonly"                                     value = "no" }
  parameter { name  = "client-output-buffer-limit-normal-hard-limit"   value = 0 }
  parameter { name  = "client-output-buffer-limit-normal-soft-limit"   value = 0 }
  parameter { name  = "client-output-buffer-limit-normal-soft-seconds" value = 0 }
  parameter { name  = "client-output-buffer-limit-pubsub-hard-limit"   value = "33554432" }
  parameter { name  = "client-output-buffer-limit-pubsub-soft-limit"   value = "8388608" }
  parameter { name  = "client-output-buffer-limit-pubsub-soft-seconds" value = 60 }
  parameter { name  = "client-output-buffer-limit-slave-soft-seconds"  value = 60 }
  parameter { name  = "close-on-slave-write"                           value = "yes" }
  parameter { name  = "cluster-enabled"                                value = "no" }
  parameter { name  = "cluster-require-full-coverage"                  value = "no" }
  parameter { name  = "databases"                                      value = 16 }
  parameter { name  = "hash-max-ziplist-entries"                       value = 512 }
  parameter { name  = "hash-max-ziplist-value"                         value = 64 }
  parameter { name  = "hll-sparse-max-bytes"                           value = 3000 }
  parameter { name  = "list-compress-depth"                            value = 0 }
  parameter { name  = "list-max-ziplist-size"                          value = -2 }
  parameter { name  = "lua-time-limit"                                 value = 5000 }
  parameter { name  = "maxclients"                                     value = 65000 }
  parameter { name  = "maxmemory-policy"                               value = "volatile-lru" }
  parameter { name  = "maxmemory-samples"                              value = 3 }
  parameter { name  = "min-slaves-max-lag"                             value = 10 }
  parameter { name  = "min-slaves-to-write"                            value = 0 }
  parameter { name  = "notify-keyspace-events"                         value = "" }
  parameter { name  = "repl-backlog-size"                              value = 1048576 }
  parameter { name  = "repl-backlog-ttl"                               value = 3600 }
  parameter { name  = "repl-timeout"                                   value = 60 }
  parameter { name  = "reserved-memory-percent"                        value = 25 }
  parameter { name  = "set-max-intset-entries"                         value = 512 }
  parameter { name  = "slave-allow-chaining"                           value = "no" }
  parameter { name  = "slowlog-log-slower-than"                        value = 10000 }
  parameter { name  = "slowlog-max-len"                                value = 128 }
  parameter { name  = "tcp-keepalive"                                  value = 300 }
  parameter { name  = "timeout"                                        value = 0 }
  parameter { name  = "zset-max-ziplist-entries"                       value = 128 }
  parameter { name  = "zset-max-ziplist-value"                         value = 64 }
}
```

```
* aws_elasticache_parameter_group.cache: Error modifying Cache Parameter Group: InvalidParameterCombination: Cannot modify more than 20 parameters in a single request
	status code: 400, request id: cab2ee8c-68db-11e7-a339-e935d4151d00
```

```
douglasgarstang@ADMINs-MacBook-Pro-2 shipwire (master) [eu1-shipwire] $ terraform -v
Terraform v0.9.10

Your version of Terraform is out of date! The latest version
is 0.9.11. You can update by downloading from www.terraform.io
```
